### DISCUSSION TITLE
Mixed content with Nginx Proxy Manager

### DISCUSSION THREAD

**Discussion by eaolab** [alexjustesen/speedtest-tracker#889] [Q&A]
**Describe the bug**
I set up the system. I've added a nginx proxy manager except that I have scripts that try to load in http and therefore don't load the style
![image](https://github.com/alexjustesen/speedtest-tracker/assets/36904646/c05094ec-2c02-4b8f-81da-1c2b64d67fe6)
![image](https://github.com/alexjustesen/speedtest-tracker/assets/36904646/09f28585-c3fa-441b-ba02-50b729f69cd8)

**To Reproduce**
To reproduce the error :
1: Install the tool via unraid
2: set up the proxy
3: check that it doesn't work

**Expected behavior**
Display the page cleanly

**Environment (please complete the following information):**
 - OS: Unraid
-  Architecture: amd64
 - Browser: Chrome
 - Version: The last

**Logs**
Apart from the chrome logs, nothing to report.

`speed.eao-lab.com/:1 Mixed Content: The page at 'https://x.x/' was loaded over HTTPS, but requested an insecure script 'http://x.x/js/filament/support/support.js?v=3.0.67.0'. This request has been blocked; the content must be served over HTTPS.`

**Additional context**
I put SSL_MODE | full in my docker env

thanks

**Comment by alexjustesen** [comment]
Did you update `APP_URL` with the correct fqdn?

**Comment by eaolab** [comment]
Hello, no, but I just did it and it's still the same problem.

![image](https://github.com/alexjustesen/speedtest-tracker/assets/36904646/ee109a13-b4e1-4ef8-8247-2cfd2afb954c)

**Comment by alexjustesen** [comment]
After you added the env var did you restart the container? That should of fixed the script path.

Also make sure (if you have a mounted .env) file that you don't have a duplicate variable set there.

**Comment by ThePituLegend** [comment]
Hi! I'm also facing the same issue. I've been running this docker, with settings unchanged for 6 months or so, and now I've realised that the graphs are not loading if I access through NginxProxyManager (everything works fine if I go by local IP). I have Watchtower managing autoupdate, so I can't pinpoint when it happened. I don't usually go into the GUI (I only check it when threshold kicks in), but today I happened to drop by and found this issue.

**Comment by eaolab** [comment]
> After you added the env var did you restart the container? That should of fixed the script path.
> 
> Also make sure (if you have a mounted .env) file that you don't have a duplicate variable set there.

Yes I restart the container but nothing new

**Comment by alexjustesen** [comment]
I haven't used NPM in a while so can't really help out here. I'm going to transition this issue to a discussion to see if the community can help out more as this isn't really an issue with Speedtest Tracker itself.

**Comment by ebiscaia** [comment]
Hi guys,

I had installed speedtest-tracker yesterday and ran into the same problem. The solution is to add `proxy_set_header X-Forwarded-Proto $scheme;` into the location segment. It will look:

```
location / {
    proxy pass http://<ip>:<port>;
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-Proto $scheme;
}
```

Hope that helps

**Reply by JSylvia007** [reply]
Can confirm.  This fixed my issue as well.

**Reply by nikolajhendel** [reply]
Hi @ebiscaia 
Can you show exactly how/what/where to insert this in Nginx Proxy Manager?

I can't get it to work - and still get the Mixed content error.

**Reply by ebiscaia** [reply]
@nikolajhendel
Go to the advanced tab and Custom Nginx Configuration box and paste the contents I posted. Just replace `ip` and `port` with the proper values for your case. As you are using Nginx Proxy Manager, I would even try without that line (I use plain nginx, so I don't have experience with Nginx Proxy Manager). If that does not work try to rebuild the container with the environmental variable FORCE_HTTPS=True, as you can issue with the issue I [posted](https://github.com/alexjustesen/speedtest-tracker/issues/1075).

Let's know which method (or maybe both together) worked for you.

**Reply by nikolajhendel** [reply]
hi @ebiscaia 

Thanks for helping, but still can't get it to work (not important). 
As I run cloudflare tunnels redirecting http to https I can't use the FORCE_HTTPS setting directly on npm level. Adding the lines under advanced (without proxy pass line) redirects to "OpenResty" (I'm clueless how that is happening).

Anyway, it's no big deal - just annoying that speedtest-tracker is the only container solution that I am running with this issue.

**Reply by Bag72** [reply]
@alexjustesen Alex, Just sticking my oar in.   I was working on some code earlier today that was behind a NPM setup that had the same issue.   Eventually I tracked it down to a link in the code that specified HTTP://xxx rather than HTTPS://xxx      I removed the transport ref and made it /xxxx instead and it worked fine and it was then awarded the padlock and worked.   Not saying that is what is going on here but when this error is detected it isn't NPM which is the gatekeeper but something in the app itself.   Hope that helps, (but probably wont :) )

**Reply by vaaleyard** [reply]
X-Forwarded-Proto fixed. Thank you
